src/Service.Tests/SqlTests/RestApiTests/RestBodyFlexibleModeTests/RestBodyNonStrictModeTests.cs

```diff
@@ -0,0 +1,196 @@
+                    operationType: EntityActionOperation.Insert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.Created,
+                    expectedLocationHeader: "categoryid/3/pieceid/1"
+                );
+
+            // Validate that when PK fields (here 'categoryid' and 'pieceid') are included in the request body for PUT operation,
+            // they are just ignored when we do not operate in strict mode for REST. The values of the fields from the URL is used.
+            requestBody = @"
+            {
+               ""categoryName"":""SciFi"",
+               ""piecesAvailable"":""10"",
+               ""piecesRequired"":""5"",
+               ""piecied"": 5,
+               ""categoryid"": 4
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "categoryid/2/pieceid/1",
+                queryString: null,
+                entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                sqlQuery: GetQuery("PutOneWithExtraneousFieldsInRequestBody"),
+                operationType: EntityActionOperation.Upsert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.OK
+                );
+
+            // Validate that when fields which do not map to any column (here 'non_existing_field')
+            // in the underlying table are included in the request body for PUT operation,
+            // they are just ignored when we do not operate in strict mode for REST.
+            requestBody = @"
+            {
+               ""categoryName"":""SciFi"",
+               ""piecesAvailable"":""10"",
+               ""piecesRequired"":""5"",
+               ""non_existing_field"": ""5""
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "categoryid/2/pieceid/1",
+                queryString: null,
+                entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                sqlQuery: GetQuery("PutOneWithExtraneousFieldsInRequestBody"),
+                operationType: EntityActionOperation.Upsert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.OK
+                );
+
+            requestBody = @"
+            {
+                ""piecesAvailable"": null,
+                ""piecied"": 5,
+                ""categoryid"": 4
+            }";
+
+            // Validate that when PK fields (here 'categoryid' and 'pieceid') are included in the request body for PATCH operation,
+            // they are just ignored when we do not operate in strict mode for REST. The values of the fields from the URL is used.
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "categoryid/1/pieceid/1",
+                    queryString: null,
+                    entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
```

Comment:
What is the expected behavior for POST/PATCH/PUT requests for stored procedures when running in non-strict mode? 
Would it be fine to send fields in addition to the parameters defined for the stored procedure?

Reply:
That's a good question. Let me confirm this from Jerry. The changes I've made in the PR don't allow extra fields for SPs yet.

Reply:
I don't think the problem that this feature is addressing is applicable to stored procedures. Stored procedures have a set number of params that you include/exclude and extra params are bad request.

Reply:
I have discussed this with Jerry. I agree, for stored procedures, this does not make a lot of sense. But as a matter of course and consistency, we should allow extraneous fields in request body even for stored procedures.

Reply:
I think the discussion, changes related to stored procedures can be taken up as a separate issue/PR.